flytekit/core/container_task.py

```diff
@@ -112,8 +114,15 @@ def _get_data_loading_config(self) -> _task_model.DataLoadingConfig:
     def _get_container(self, settings: SerializationSettings) -> _task_model.Container:
         env = settings.env or {}
         env = {**env, **self.environment} if self.environment else env
+        if isinstance(self._image, ImageSpec):
+            if settings.fast_serialization_settings is None or not settings.fast_serialization_settings.enabled:
```

Comment:
should we use [ get_registerable_container_image](https://github.com/ByronHsu/flytekit/blob/5113833811c9e7eedfdcd8231fc55a5cf913ecd4/flytekit/core/python_auto_container.py#L265) to resolve the image name so it can also handle the case like `{{.image.default.fqn}}:{{.image.default.version}})`

Reply:
done